package.json

```
@@ -3,37 +3,53 @@
   "version": "7.0.0-beta.0",
   "description": "The router for easy microfrontends",
   "type": "module",
-  "main": "lib/umd/single-spa.min.cjs",
+  "main": "./lib/umd/single-spa.min.cjs",
+  "types": "./lib/umd/types/single-spa.d.cts",
+  "module": "./lib/esm/single-spa.min.js",
   "publishConfig": {
     "access": "public"
   },
   "exports": {
     ".": {
       "development": {
-        "import": "./lib/esm/single-spa.dev.js",
-        "require": "./lib/umd/single-spa.dev.cjs",
-        "types": "./typings/single-spa.d.ts"
+        "import": {
+          "types": "./lib/esm/types/single-spa.d.ts",
+          "default": "./lib/esm/single-spa.min.js"
+        },
+        "require": {
+          "types": "./lib/umd/types/single-spa.d.cts",
+          "default": "./lib/umd/single-spa.min.cjs"
+        }
       },
       "production": {
-        "import": "./lib/esm/single-spa.min.js",
-        "require": "./lib/umd/single-spa.min.cjs",
-        "types": "./typings/single-spa.d.ts"
+        "import": {
+          "types": "./lib/esm/types/single-spa.d.ts",
+          "default": "./lib/esm/single-spa.min.js"
+        },
+        "require": {
+          "types": "./lib/umd/types/single-spa.d.cts",
+          "default": "./lib/umd/single-spa.min.cjs"
+        }
       },
       "default": {
-        "import": "./lib/esm/single-spa.min.js",
-        "require": "./lib/umd/single-spa.min.cjs",
-        "types": "./typings/single-spa.d.ts"
+        "import": {
+          "types": "./lib/esm/types/single-spa.d.ts",
+          "default": "./lib/esm/single-spa.min.js"
+        },
+        "require": {
+          "types": "./lib/umd/types/single-spa.d.cts",
+          "default": "./lib/umd/single-spa.min.cjs"
+        }
       }
     },
     "./package.json": "./package.json",
     "./lib/*": "./lib/*"
   },
   "scripts": {
-    "build": "pnpm run clean && concurrently pnpm:build:dev pnpm:build:prod pnpm:build:types",
+    "build": "pnpm run clean && concurrently pnpm:build:dev pnpm:build:prod && node ./scripts/remove-extraneous-types.cjs",
```

rimraf cli can do what remove-extraneous-types.cjs is doing

```suggestion
    "build": "pnpm run clean && concurrently pnpm:build:dev pnpm:build:prod && rimraf ./lib/umd/types-tsc ./lib/esm/types-tsc",
```

---

Done.